doc/Overview/Serialization/SerializationOverview.md

```diff
@@ -1,43 +1,133 @@
 
-public App()
+```csharp
+public class MyViewModel : ObservableObject
 {
-    Host = UnoHost
-        .CreateDefaultBuilder()
-        .UseSerialization()
-        .Build();
-    // ........ //
+    private ISerializer _serializer;
+
+    public MyViewModel(ISerializer serializer)
+    {
+        this._serializer = serializer;
+    }
+
+    public void Serialize()
+    {
+        var myObject = new Person();
+        var json = _serializer.Serialize(myObject);
+    }
 }
 ```
 
-Access `ISerializer` to serialize and deserialize json data
-
 ## Configuring ISerializer
 
-The default serializer implementation uses System.Text.Json. The serialization can be configured by registering an instance of JsonSerializerOptions
+### JSON
+
+The default serializer implementation only supports serializing to JSON. Because it uses `System.Text.Json`, the specifics of this behavior can be configured with `JsonSerializerOptions`. The following example shows how to register an instance of `JsonSerializerOptions` with the host.
+
+```csharp
+protected override void OnLaunched(LaunchActivatedEventArgs e)
+{
+    var appBuilder = this.CreateBuilder(args)
+        .Configure(host => {
+            host
+            .UseSerialization(services =>
+            {
+                services
+                    .AddSingleton(new JsonSerializerOptions { PropertyNameCaseInsensitive = true });
+            });
+        });
+...
+```
+
+#### Source Generation
+
+As of .NET 6, a code generation-enabled serializer is supported. The type to serialize is named `Person` in this example.
+
+```csharp
+public class Person
+{
+    public Person() { }
+
+    public Person(string name, int age, double height, double weight)
+    {
+        Name = name;
+        Age = age;
+        Height = height;
+        Weight = weight;
+    }
+    
+    public string? Name { get; set; }
+    public int Age { get; set; }
+    public double  Height { get; set; }
+    public double Weight { get; set; }
+}
+```
+
+To leverage the source generation feature for JSON serialization in an Uno.Extensions application, define a partial class which derives from a `JsonSerializerContext`, and specify which type is serializable with the `JsonSerializable` attribute:
 
 ```csharp
-private IHost Host { get; }
+using System.Text.Json.Serialization;
 
-public App()
+[JsonSerializable(typeof(Person))]
+public partial class PersonJsonContext : JsonSerializerContext
 {
-    Host = UnoHost
-        .CreateDefaultBuilder()
-        .UseSerialization()
-        .ConfigureServices(services =>
-        {
-            services
-                .AddSingleton(new JsonSerializerOptions { PropertyNameCaseInsensitive = true });
-        }
-        .Build();
-    // ........ //
 }
 ```
+
+The `PersonJsonContext` class is then used to serialize and deserialize instances of `Person`:
+
+```csharp
+var person = new Person("Megan", 23, 1.8, 90.0);
+
+var json = JsonSerializer.Serialize(person!, PersonJsonContext.Default.Person);
```

Comment:
The whole point is to use ISerializer as part of Serizalization. You need to register the PersonJsonContext.Default as part of app builder. See https://github.com/unoplatform/uno.extensions/blob/18f534042d9204ea3ad5962fbc517903c953902b/samples/Playground/Playground.Shared/AppHost.cs#L35-L38